src/anomalib/utils/metrics/perimg/plot.py

```diff
@@ -474,3 +474,328 @@ def _sort_legend(handles: list, labels: list[str]):
+            color="cyan",
+            alpha=0.2,
+        ),
+    ]
+
+    if ubound is not None:
+        handles.append(
+            ax.axvline(
+                ubound,
+                label=f"Upper Bound ({float(100 * ubound):.2g}%)",
+                linestyle="--",
+                color="black",
+            )
+        )
+
+    if lbound is not None:
+        handles.append(
+            ax.axvline(
+                lbound,
+                label=f"Lower Bound ({float(100 * lbound):.2g}%)",
+                linestyle="--",
+                color="gray",
+            )
+        )
+
+    ax.legend(
+        handles,
+        [h.get_label() for h in handles],
+        title="AUC integration",
+        loc="center right",
+        fontsize="small",
+        title_fontsize="small",
+    )
+
+    if current_legend is not None:
+        ax.add_artist(current_legend)
+
+
+# =========================================== PImFPR ===========================================
+
+
+def plot_pimfpr_curves_norm_vs_anom(
+    fprs: Tensor,
+    shared_fpr: Tensor,
+    image_classes: Tensor,
+    ax: Axes | None = None,
+) -> tuple[Figure | None, Axes]:
+    """Plot shared FPR vs in-image FPR curves, normal image (in blue) VS anomalous images (in red).
+
+    Args:
+        ax: matplotlib Axes
+        shared_fpr: shape (num_thresholds,)
+        fprs: shape (num_images, num_thresholds)
+        image_classes: shape (num_images,)
+    """
+
+    # ** validate ** [GENERIC]
+    _validate_rate_curve(shared_fpr)
+    _validate_perimg_rate_curves(fprs, nan_allowed=True)  # anomalous images may have `nan`s if all pixels are anomalous
+    _validate_image_classes(image_classes)
+
+    if fprs.shape[0] != image_classes.shape[0]:
+        raise ValueError(
+            f"Expected argument `fprs` to have the same number of images as argument `image_classes`, "
+            f"but got {fprs.shape[0]} images and {image_classes.shape[0]} images, respectively."
+        )
+
+    # there may be `nan`s but only in the anomalous images
+    # in the curves of normal images, there should NOT be `nan`s
+    if (image_classes == 0).any():
+        _validate_perimg_rate_curves(fprs[image_classes == 0], nan_allowed=False)
+
+    # ** validate ** [SPECIFIC]
+    # it's a normal vs. anomalous plot, so there should be at least one of each
+    _validate_atleast_one_anomalous_image(image_classes)
+    _validate_atleast_one_normal_image(image_classes)
+
+    fig, ax = plt.subplots(figsize=(7, 7)) if ax is None else (None, ax)
+
+    # ** plot **
+    kwargs_perimg = [
+        dict(
+            # color the lines by the image class; normal = blue, anomalous = red
+            color="blue" if img_cls == 0 else "red",
+            # make the legend only show one normal and one anomalous line by passing `label=None`
+            # and just modifying the first of each class in the lines below
+            label=None,
+        )
+        for imgidx, img_cls in enumerate(image_classes)
```

Comment:
Shall we add the changes

- `thidx` by `thresh_idx`
- `img`/`imgidx` by `image`/`image_idx`

to the list of renames in the refactor?

because more of these are coming up in the next PRs :grimacing: